modular_skyrat/modules/reagent_forging/code/crafting_bench.dm

```diff
@@ -0,0 +1,281 @@
+	name = "forging work bench"
+	desc = "A crafting bench used to create forging items."
+	icon = 'modular_skyrat/modules/reagent_forging/icons/obj/forge_structures.dmi'
+	icon_state = "crafting_bench_empty"
+
+	anchored = TRUE
+	density = TRUE
+
+	///the current goal item that is obtainable
+	var/goal_item_path
+	///the name of the goal item
+	var/goal_name
+	///the amount of chains within the bench
+	var/chain_amount = 0
+	///the amount of chains required
+	var/required_chain = 0
+	///the amount of plates within the bench
+	var/plate_amount = 0
+	///the amount of plates required
+	var/required_plate = 0
+	///the amount of coils within the bench
+	var/coil_amount = 0
+	///the amount of coils required
+	var/required_coil = 0
+	///the amount of hits required to complete the item
+	var/required_hits = 0
+	///the current amount of hits
+	var/current_hits = 0
+	///the amount of wood currently stored
+	var/current_wood = 0
+	//so we can't just keep being hit without cooldown
+	COOLDOWN_DECLARE(hit_cooldown)
+	///the choices allowed in crafting
+	var/list/allowed_choices = list(
+		"Chain Helmet",
+		"Chain Armor",
+		"Chain Gloves",
+		"Chain Boots",
+		"Plated Boots",
+		"Horseshoes",
+		"Ring",
+		"Collar",
+		"Handcuffs",
+		"Borer Cage",
+		"Pavise Shield",
+		"Buckler Shield",
+		"Coil",
+	)
+	///the paths of the allowed choices
+	var/static/list/choice_paths = list(
+		/obj/item/clothing/head/helmet/reagent_clothing,
+		/obj/item/clothing/suit/armor/reagent_clothing,
+		/obj/item/clothing/gloves/reagent_clothing,
+		/obj/item/clothing/shoes/chain_boots,
+		/obj/item/clothing/shoes/plated_boots,
+		/obj/item/clothing/shoes/horseshoe,
+		/obj/item/clothing/gloves/ring/reagent_clothing,
+		/obj/item/clothing/neck/kink_collar/reagent_clothing,
+		/obj/item/restraints/handcuffs/reagent_clothing,
+		/obj/item/cortical_cage,
+		/obj/item/shield/riot/buckler/reagent_weapon/pavise,
+		/obj/item/shield/riot/buckler/reagent_weapon,
+		/obj/item/forging/coil,
+	)
+
+/obj/structure/reagent_crafting_bench/examine(mob/user)
+	. = ..()
+	if(goal_name)
+		. += span_notice("Goal Item: [goal_name]")
+		. += span_notice("When you have the necessary materials, begin hammering!<br>")
+		if(required_chain)
+			. += span_warning("[chain_amount]/[required_chain] chains stored.")
+		if(required_plate)
+			. += span_warning("[plate_amount]/[required_plate] plates stored.")
+		if(required_coil)
+			. += span_warning("[coil_amount]/[required_coil] coils stored.")
+	if(length(contents))
+		. += span_notice("Held Item: [contents[1]]")
+
+/obj/structure/reagent_crafting_bench/update_appearance(updates)
+	. = ..()
+	icon_state = "crafting_bench_[length(contents) ? "full" : "empty"]"
+
+//when picking a design or clearing a design
+/obj/structure/reagent_crafting_bench/attack_hand(mob/living/user, list/modifiers)
+	. = ..()
+	if(length(contents))
+		var/obj/item/moving_item = contents[1]
+		user.put_in_hands(moving_item)
+		balloon_alert(user, "item retrieved!")
```

Comment:
this isn't at all a you issue, but it annoys me how balloon alerts aren't consistent in if the first letter is capitalized or not throughout the code

Reply:
yea :/